torch_geometric/datasets/generators/graph_generator.py

```diff
@@ -0,0 +1,57 @@
+from typing import Callable, Optional
+
+import torch
+
+from torch_geometric.data import Data
+
+
+class GraphGenerator:
+    def __init__(self, motif: Optional[Callable], num_nodes: int = 300):
+        self.num_nodes = num_nodes
+        self.motif = motif
+        self.edge_index = None
+        self.edge_label = None
+        self.expl_mask = None
+        self.node_label = None
+        self.x = None
+
+    def generate_base_graph(self) -> Data:
+        raise NotImplementedError
+
+    # TODO: FeatureGenerator (feature distribution...)
+    def generate_feature(self, num_features: int = 10):
+        self.x = torch.ones((self.num_nodes, num_features), dtype=torch.float)
+
+    def attach_motif(self, num_motifs=80,
+                     connection_distribution: str = 'random'):
+        if connection_distribution == 'random':
+            connecting_nodes = torch.randperm(self.num_nodes)[:num_motifs]
+        else:
+            step = self.num_nodes // num_motifs
+            connecting_nodes = torch.arange(0, self.num_nodes, step)
```

Comment:
there's also a question of whether the attachment will be made on the same orbit of the motif (e.g. always the tip of house or a random node in house motif). Either way is ok for this PR, but might be a good idea to document it.

In future, if we see request to have the other option, we could have a different pr to address that.

Reply:
I added the description in the classes but I'm planning to work more on writing better documentation on those.